src/Model/Scope/Condition.php

```diff
@@ -0,0 +1,384 @@
+
+                    $field = array_pop($references);
+
+                    foreach ($references as $link) {
+                        $model = $model->refLink($link);
+                    }
+
+                    // '#' -> has # referenced records
+                    // '?' -> has any referenced records
+                    // '!' -> does not have any referenced records
+                    if (in_array($field, ['#', '!', '?'], true)) {
+                        // if no operator consider '#' as 'any record exists'
+                        if ($field == '#' && !$operator) {
+                            $field = '?';
+                        }
+
+                        if (in_array($field, ['!', '?'], true)) {
+                            $operator = '=';
+                            $value = $field == '?' ? 1 : 0;
+                        }
+                    } else {
+                        // otherwise add the condition to the referenced model
+                        // and check if any records exist matching the criteria
+                        $model->addCondition($field, $operator, $value);
+                        $operator = '=';
+                        $value = 1;
+                    }
+
+                    // if not counting we check for existence only
+                    $field = $field == '#' ? $model->action('count') : $model->action('exists');
+                } else {
+                    $field = $model->getField($field);
+                }
+            }
+
+            // @todo: value is array
+            // convert the value using the typecasting of persistence
+            if ($field instanceof Field && $model->persistence && !in_array(strtoupper((string) $operator), self::$skipValueTypecast, true)) {
+                $value = $model->persistence->typecastSaveField($field, $value);
+            }
+
+            // only expression contained in $field
+            if (!$operator) {
+                return [$field];
+            }
+
+            // skip explicitly using '=' as in some cases it is transformed to 'in'
+            // for instance in dsql so let exact operator be handled by Persistence
+            if ($operator === '=') {
+                return [$field, $value];
+            }
+        }
+
+        return [$field, $operator, $value];
+    }
+
+    public function isEmpty(): bool
+    {
+        return array_filter([$this->key, $this->operator, $this->value]) ? false : true;
+    }
+
+    public function validate(array $values): array
+    {
+        if (!$this->isActive()) {
+            return [];
+        }
+
+        if (!$model = $this->model) {
+            throw new Exception('Model must be set using setModel to validate');
+        }
+
+        $data = [1 => $values];
+
+        $model->withPersistence(new Array_($data));
+
+        $model->add($this);
+
+        return count($model->export()) !== 0 ? [] : [$this];
+    }
+
+    public function negate()
```

Comment:
This is tough - it does not negate the condition because of NULL will never match with `<` or `>`.